sdk/tables/data-tables/src/TableClient.ts

```diff
@@ -549,25 +550,43 @@ export class TableClient {
     }
   }
 
-  /**
-   * Creates a new Batch to collect sub-operations that can be submitted together via submitBatch
-   * @param partitionKey - partitionKey to which the batch operations will be targetted to
-   */
-  // eslint-disable-next-line @azure/azure-sdk/ts-naming-options
-  public createBatch(partitionKey: string): TableBatch {
-    const batchId = Uuid.generateUuid();
+  public async submitTransaction(actions: TransactionAction[]): Promise<TableTransactionResponse> {
+    const partitionKey = actions[0][1].partitionKey;
+    const transactionId = Uuid.generateUuid();
     const changesetId = Uuid.generateUuid();
-    const innerBatchRequest = createInnerBatchRequest(batchId, changesetId);
-    const internalClientOptions: InternalBatchClientOptions = { innerBatchRequest };
+    const innerTransactionRequest = createInnerTransactionRequest(transactionId, changesetId);
+    const internalClientOptions: InternalTransactionClientOptions = {
+      innerTransactionRequest: innerTransactionRequest
+    };
     const interceptClient = new TableClient(this.url, this.tableName, internalClientOptions);
```

Comment:
I think this works ok for now functionally, but it seems heavy to re-create a `TableClient` instance each time. Can we cache/re-use this across operations, maybe by clearing the state out after submitting?

Reply:
Yeah, that makes sense. I'll cache the interceptClient so we just create it once and reuse it. We don't need to clear the state for this one. InternalTableTransaction would need to be cleared, however, most of the work of that class is to keep the state, so I think it would be safer to just re-create it on every submit.